nearcore/src/lib.rs

```diff
@@ -74,8 +74,78 @@ pub fn get_default_home() -> PathBuf {
     PathBuf::default()
 }
 
+mod r#impl {
```

Comment:
can you explain why like this (and not putting the struct directly  ? 

(or optionally moving this struct to a separate file)

Reply:
I was thinking of moving it out to a separate file but there were not many files so I thought it's intended to have only several files.

This is needed so the struct can't be created directly but only using ::new method.